src/Bicep.Core/Emit/EmitConstants.cs

```diff
@@ -16,6 +17,9 @@ public static class EmitConstants
 
     public const string NestedDeploymentResourceApiVersionWithIdentityInModulesSupport = "2025-04-01";
 
+    public static readonly FrozenSet<string> ResourceInfoProperties
+        = new[] { "id", "name", "type", "apiVersion" }.ToFrozenSet();
```

Comment:
Is there somewhere else we can reference these from rather than duplicating?

Reply:
I moved this to `EmitConstants` from a private static member of a visitor in the same namespace. It could go in `LanguageConstants` instead, but this seemed like a reasonable place since whether an expression compiles to `resourceInfo(...)` or `reference(...)` shouldn't be a concern of the core compiler.